### PR TITLE
【BIT】引擎修复：Update all outdated rules

### PR DESCRIPTION
**Pull request by cangtianhuang**
## 修改内容

1. 更新本人撰写的所有过时 Rule，使用 Code 数据类代码划分
2. 废弃 mapping.json 中的 composition_steps 配置（过于局限，仅有 add_n 使用且未完全适配），移除 read_mapping、GenericRule 中相关内容
3. 更新 Paddle2Torch README
4. 修复了 `paddle.add_n` ，为其注册并编写 AddNRule
5. 修复了 `paddle.Tensor.__and__` ，其在 https://github.com/PFCCLab/PaddleAPITest/pull/107 后无法 pass
6. 修复了 `paddle.nn.functional.channel_shuffle` 的 DataFormatRule，其在 https://github.com/PFCCLab/PaddleAPITest/pull/103 中被错误修改
7. 移动两条已在 fresh_report 中记录的 accuracy error 配置

## 测试结果

所有常规 paddle.add_n 配置均通过 accuracy 测试，原先的两条 paddleonly 配置被支持，合并移入 support2torch_1 ；两条 amp 配置移入 support2torch_amp 中：
image

所有 paddle.nn.functional.channel_shuffle 配置通过 accuracy 测试；paddle.Tensor.set_、paddle.cumprod 的两条配置被移入 accuracy_error_1

### CODE REVIEW

**Review by yuwu46 — APPROVED** [review]
LGTM

File: tester/api_config/api_config_accuracy_error_1.txt [hunk 1]
```diff
@@ -73,3 +73,5 @@ paddle.nn.functional.max_pool2d(Tensor([8, 32, 272, 272],"float32"), kernel_size
 paddle.nn.functional.max_pool2d(Tensor([8, 48, 224, 224],"float32"), kernel_size=2, stride=1, padding="SAME", return_mask=False, ceil_mode=True, data_format="NCHW", name=None, )
 paddle.nn.functional.max_pool2d(Tensor([8, 512, 20, 20],"float16"), kernel_size=13, stride=1, padding=6, return_mask=False, ceil_mode=False, data_format="NCHW", name=None, )
 paddle.nn.functional.max_pool3d(x=Tensor([2, 3, 8, 8, 8],"float64"), kernel_size=list[3,3,3,], stride=list[3,2,1,], padding=list[1,2,1,], )
+paddle.cumprod(x=Tensor([3, 4],"int32"), dim=0, )
+paddle.Tensor.set_(Tensor([3, 8],"float16"), Tensor([6, 3],"float16"), list[3,8,], list[2,2,], 0, )
```
Comment (yuwu46) [inline tester/api_config/api_config_accuracy_error_1.txt:77]: 移入accuracy_error后，是不是还需要把log文件加入fresh_reporter里？

Comment (yuwu46) [inline tester/api_config/api_config_accuracy_error_1.txt:77]: 移入accuracy_error后，是不是还要把log放在fresh_reporter里？

Comment (cangtianhuang) [reply tester/api_config/api_config_accuracy_error_1.txt:77]: 之前已经在 fresh_reporter 记录过，只是被错误地移入 support2torch 中了，在昨天的全量测试中将这两个排查出来了

**Review by yuwu46 — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM
